update releasing doc on when to wait (#1485)
medic/cht-docs: content/en/contribute/code/releasing/_index.md

{{% alert title="Info" %}} You can find the versions currently supported, dependencies, and release notes for the CHT Core 
 [on the Releases page]({{% ref "core/releases" %}}). {{% /alert %}}

### When to wait to release

> In general, we do not want to wait to do a release.  The moment the release manager feels the release is ready, they should release it.  This may even be with just one ticket! 

Reviewer: [Per the guidelines](https://docs.communityhealthtoolkit.org/contribute/docs/style-guide/#using-we), using `we` can be confusing for the reader. There are several mentions of it in this new section and I suggest we rephrase it to avoid the `we`/`us` confusion. 

```suggestion
In general, release managers should not feel they have to wait to do a release. The moment the release manager feels the release is ready, they should release it. This may even be with just one ticket! 
```

> There are two conditions that we would wait:

Reviewer: ```suggestion
There are two scenarios where a release manager should wait before doing the release:
```

> 1. There is a new bug found in `master` that has not been released.  This should be fixed before the release.

Reviewer 2: The wording here feels not specific enough. What about this?

```suggestion
1. There is a bug found in `master` that did not exist in the previous release.  This should be fixed before the release.
```

> 1. There is a bug found in an existing release that can be fixed in 5 days.  In less than 5 days the fixed code will be released.  In more than 5 days, the bug will be released and a new release will ship later with the fix for the bug.

Reviewer 2: ```suggestion
1. There is a high priority bug found in an existing release that we think can be fixed in less than 5 days. The way to determine if a bug is high priority is if we would release a service pack to fix it later. If so then it makes sense to wait for the fix to save us the effort of having to do two releases in quick succession. If it's going to take longer than 5 days then it's worth the additional effort of releasing a service pack. After 5 days if the fix isn't merged, then continue with the release regardless.
```

Author: taking Andra's revision in 20da0fce

> The reason we have these rules around when we wait, is so that everyone knows what they are and the release manager never needs to ask "should we wait?" - they'll know already!

Reviewer 2: ```suggestion
The reason we have these rules around when we wait, is so that everyone knows what they are and the release manager never needs to ask "should we wait?" - they'll know already!

While waiting the release manager should use the time effectively by doing as many release steps as possible including writing release notes, preparing the release branch, etc so that the release can be finalized as quickly as possible once the fixes are made.
```

Reviewer: Same here, please replace the mention of `we` with "release manager" or something more generic.

Author: took both in 35a099a8  - thanks!
